app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,22 @@
+class CommentsController < ApplicationController
+  def create
+    user = current_user
+    comment = Comment.new(post_params)
+    if comment.save
+      redirect_to user_post_path(user_id: user.id, id: params[:post_id])
+    else
+      redirect_to user_post_path(user_id: user.id, id: params[:post_id]), alert: comment.errors.full_messages.first
+    end
+  end
+
+  private
+
+  def post_params
+    author = current_user
+    post = Post.find(params[:post_id])
+    hash = params.require(:comment).permit(:text)
+    hash[:author] = author
+    hash[:post] = post
+    hash
+  end
```

Comment:
[OPTIONAL] There's an easier way to achieve this. Kindly consider using this code for comment parameters.

```
  private

  def comment_params
    params.require(:comment).permit(:text)
  end
```

Reply:
I completely forgot to refactor this piece of code. Thank you so much for pointing to it!